pkg/rke2/rke2.go

```diff
@@ -297,6 +302,29 @@ func startContainerd(dataDir string, errChan chan error, cmd *exec.Cmd) {
 	errChan <- cmd.Run()
 }
 
+func waitForContainerd(ctx context.Context) error {
+	first := true
+	for {
+		conn, err := containerdk3s.CriConnection(ctx, containerdSock)
```

Comment:
We just keep looping until it does not return an error.

This is essentially copypasta from k3s; I suppose I could move it into a dedicated function on that side and then just call it here rather than having a second copy of it.
https://github.com/k3s-io/k3s/blob/fc8309a46c1e2a1c9e2491b5c63fe3c9a63ab6cb/pkg/agent/containerd/containerd.go#L97-L115

Reply:
Will move into a function on the K3s side and just call it from here